File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        p1 = dummy
        for _ in range(left - 1):
            p1 = p1.next
        p2 = p1.next
        for _ in range(left, right):
            tmp = p2.next
            p2.next = tmp.next
            tmp.next = p1.next
            p1.next = tmp
        return dummy.next

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} left
 * @param {number} right
 * @return {ListNode}
 */
var reverseBetween = function(head, left, right) {
  const dummy = new ListNode(-1, head);
  let prev = dummy;
  
  for (let i = 0; i < left - 1; i++) prev = prev.next;
  
  let second = prev.next;
  for (let i = left; i < right; i++) {
    const next = second.next;
    second.next = next.next;
    next.next = prev.next;
    prev.next = next;
  }
  
  return dummy.next;
};